From 4c37a298152fdd2195482b4dbe01f7733554a253 Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Mon, 19 Sep 2011 14:12:34 +0800
Subject: [PATCH 184/635] ENGR00153881 HDMI: need replug-in the HDMI cable to
 enable auto config

Update framebuffer HAL to forcely write video mode

Signed-off-by: xianzhong <b07117@freescale.com>
---
 libgps/gps.c                      | 25 ++++++++++++-------------
 mx5x/hwcomposer/BG_device.cpp     | 15 +++++++++++++++
 mx5x/hwcomposer/FG_device.cpp     | 16 +++++++++++++++-
 mx5x/hwcomposer/blit_gpu.cpp      | 16 +++++++++++++++-
 mx5x/hwcomposer/blit_gpu.h        | 16 ++++++++++++++++
 mx5x/hwcomposer/blit_ipu.cpp      | 15 +++++++++++++++
 mx5x/hwcomposer/blit_ipu.h        | 16 ++++++++++++++++
 mx5x/hwcomposer/hwc_common.cpp    | 15 +++++++++++++++
 mx5x/hwcomposer/hwc_common.h      | 15 +++++++++++++++
 mx5x/hwcomposer/hwcomposer.cpp    |  4 ++--
 mx5x/hwcomposer/output_device.cpp | 15 +++++++++++++++
 mx5x/libgralloc/framebuffer.cpp   | 16 ++++++++++------
 mx6/hwcomposer/BG_device.cpp      | 15 +++++++++++++++
 mx6/hwcomposer/FG_device.cpp      | 16 ++++++++++++++++
 mx6/hwcomposer/blit_gpu.cpp       | 15 +++++++++++++++
 mx6/hwcomposer/blit_gpu.h         | 17 +++++++++++++++++
 mx6/hwcomposer/blit_ipu.cpp       | 16 ++++++++++++++++
 mx6/hwcomposer/blit_ipu.h         | 17 +++++++++++++++++
 mx6/hwcomposer/hwc_common.cpp     | 16 ++++++++++++++++
 mx6/hwcomposer/hwc_common.h       | 16 ++++++++++++++++
 mx6/hwcomposer/hwcomposer.cpp     |  4 ++--
 mx6/hwcomposer/output_device.cpp  | 16 ++++++++++++++++
 22 files changed, 307 insertions(+), 25 deletions(-)

diff --git a/libgps/gps.c b/libgps/gps.c
index 4163922..c4eb6e5 100644
--- a/libgps/gps.c
+++ b/libgps/gps.c
@@ -3,19 +3,18 @@
  */
 
 /*
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
-
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
-
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
  */
 
  /*
diff --git a/mx5x/hwcomposer/BG_device.cpp b/mx5x/hwcomposer/BG_device.cpp
index eb535b3..7826449 100755
--- a/mx5x/hwcomposer/BG_device.cpp
+++ b/mx5x/hwcomposer/BG_device.cpp
@@ -1,3 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 #include <sys/mman.h>
 #include <sys/ioctl.h>
diff --git a/mx5x/hwcomposer/FG_device.cpp b/mx5x/hwcomposer/FG_device.cpp
index 1850fe5..8b4ab5b 100755
--- a/mx5x/hwcomposer/FG_device.cpp
+++ b/mx5x/hwcomposer/FG_device.cpp
@@ -1,4 +1,18 @@
-
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 #include <hardware/hardware.h>
 
diff --git a/mx5x/hwcomposer/blit_gpu.cpp b/mx5x/hwcomposer/blit_gpu.cpp
index 4fc2421..37e4a78 100755
--- a/mx5x/hwcomposer/blit_gpu.cpp
+++ b/mx5x/hwcomposer/blit_gpu.cpp
@@ -1,4 +1,18 @@
-
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 #include <hardware/hardware.h>
 
diff --git a/mx5x/hwcomposer/blit_gpu.h b/mx5x/hwcomposer/blit_gpu.h
index ed8f5d9..f191e68 100755
--- a/mx5x/hwcomposer/blit_gpu.h
+++ b/mx5x/hwcomposer/blit_gpu.h
@@ -1,3 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #ifndef _BLIT_GPU_H_
 #define _BLIT_GPU_H_
 
diff --git a/mx5x/hwcomposer/blit_ipu.cpp b/mx5x/hwcomposer/blit_ipu.cpp
index 53f611f..863b126 100755
--- a/mx5x/hwcomposer/blit_ipu.cpp
+++ b/mx5x/hwcomposer/blit_ipu.cpp
@@ -1,3 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 
 #include <hardware/hardware.h>
diff --git a/mx5x/hwcomposer/blit_ipu.h b/mx5x/hwcomposer/blit_ipu.h
index b12114f..15d163e 100755
--- a/mx5x/hwcomposer/blit_ipu.h
+++ b/mx5x/hwcomposer/blit_ipu.h
@@ -1,3 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #ifndef _BLIT_IPU_H_
 #define _BLIT_IPU_H_
 
diff --git a/mx5x/hwcomposer/hwc_common.cpp b/mx5x/hwcomposer/hwc_common.cpp
index ec2a88b..f64fbda 100755
--- a/mx5x/hwcomposer/hwc_common.cpp
+++ b/mx5x/hwcomposer/hwc_common.cpp
@@ -1,3 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 
 #include <hardware/hardware.h>
diff --git a/mx5x/hwcomposer/hwc_common.h b/mx5x/hwcomposer/hwc_common.h
index 60359bd..a1576e0 100755
--- a/mx5x/hwcomposer/hwc_common.h
+++ b/mx5x/hwcomposer/hwc_common.h
@@ -1,3 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 #ifndef _HWC_FSL_H_
 #define _HWC_FSL_H_
diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index 26d51b3..a0b6c57 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -1,6 +1,4 @@
 /*
- * Copyright (C) 2010 The Android Open Source Project
- *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
@@ -14,6 +12,8 @@
  * limitations under the License.
  */
 
+/*Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #include <hardware/hardware.h>
 #include <hardware/overlay.h>
 
diff --git a/mx5x/hwcomposer/output_device.cpp b/mx5x/hwcomposer/output_device.cpp
index 0022305..cdbbdff 100755
--- a/mx5x/hwcomposer/output_device.cpp
+++ b/mx5x/hwcomposer/output_device.cpp
@@ -1,3 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 #include <hardware/hardware.h>
 
diff --git a/mx5x/libgralloc/framebuffer.cpp b/mx5x/libgralloc/framebuffer.cpp
index 4e3441b..03eeffb 100755
--- a/mx5x/libgralloc/framebuffer.cpp
+++ b/mx5x/libgralloc/framebuffer.cpp
@@ -774,7 +774,7 @@ static int set_graphics_fb_mode(int fb, int dual_disp)
 
     fp_cmd = open("/proc/cmdline",O_RDONLY, 0);
     if(fp_cmd < 0) {
-        LOGI("Error! Cannot open /proc/cmdline");
+        LOGI("Error %d! Cannot open /proc/cmdline", fp_cmd);
         goto set_graphics_fb_mode_error;
     }
 
@@ -793,7 +793,7 @@ static int set_graphics_fb_mode(int fb, int dual_disp)
     sprintf(temp_name, "/sys/class/graphics/fb%d/modes", fb);
     fp_modes = open(temp_name,O_RDONLY, 0);
     if(fp_modes < 0) {
-        LOGI("Error! Cannot open %s", temp_name);
+        LOGI("Error %d! Cannot open %s", fp_modes, temp_name);
         goto set_graphics_fb_mode_error;
     }
 
@@ -821,21 +821,25 @@ static int set_graphics_fb_mode(int fb, int dual_disp)
     sprintf(temp_name, "/sys/class/graphics/fb%d/mode", fb);
     fp_mode = open(temp_name,O_RDWR, 0);
     if(fp_mode < 0) {
-        LOGI("Error! Cannot open %s", temp_name);
+        LOGI("Error %d! Cannot open %s", fp_mode, temp_name);
         goto set_graphics_fb_mode_error;
     }
 
     memset(fb_mode, 0, sizeof(fb_mode));
     size = read(fp_mode, fb_mode, sizeof(fb_mode));
-    if(size <= 0)
+    if(size < 0)
     {
         LOGI("Error! Cannot read %s", temp_name);
         goto set_graphics_fb_mode_error;
     }
 
-    if(strncmp(fb_mode, disp_mode, strlen(fb_mode)))
+    if(strncmp(fb_mode, disp_mode, strlen(disp_mode)+1))
     {
-        write(fp_mode, disp_mode, strlen(disp_mode)+1);
+        size = write(fp_mode, disp_mode, strlen(disp_mode)+1);
+		if(size <= 0)
+	    {
+	        LOGI("Error! Cannot write %s", temp_name);
+	    }
     }
 
     close(fp_mode); fp_mode = 0;
diff --git a/mx6/hwcomposer/BG_device.cpp b/mx6/hwcomposer/BG_device.cpp
index 6b6a067..4a5de6c 100755
--- a/mx6/hwcomposer/BG_device.cpp
+++ b/mx6/hwcomposer/BG_device.cpp
@@ -1,3 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 #include <sys/mman.h>
 #include <sys/ioctl.h>
diff --git a/mx6/hwcomposer/FG_device.cpp b/mx6/hwcomposer/FG_device.cpp
index 784c5c1..824d368 100755
--- a/mx6/hwcomposer/FG_device.cpp
+++ b/mx6/hwcomposer/FG_device.cpp
@@ -1,4 +1,20 @@
 
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 
 #include <hardware/hardware.h>
 
diff --git a/mx6/hwcomposer/blit_gpu.cpp b/mx6/hwcomposer/blit_gpu.cpp
index d7b81c5..e348c9d 100755
--- a/mx6/hwcomposer/blit_gpu.cpp
+++ b/mx6/hwcomposer/blit_gpu.cpp
@@ -1,3 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
 
 
 #include <hardware/hardware.h>
diff --git a/mx6/hwcomposer/blit_gpu.h b/mx6/hwcomposer/blit_gpu.h
index 04e86f0..15c1860 100755
--- a/mx6/hwcomposer/blit_gpu.h
+++ b/mx6/hwcomposer/blit_gpu.h
@@ -1,3 +1,20 @@
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #ifndef _BLIT_GPU_H_
 #define _BLIT_GPU_H_
 
diff --git a/mx6/hwcomposer/blit_ipu.cpp b/mx6/hwcomposer/blit_ipu.cpp
index 24c0fdd..72625ce 100755
--- a/mx6/hwcomposer/blit_ipu.cpp
+++ b/mx6/hwcomposer/blit_ipu.cpp
@@ -1,4 +1,20 @@
 
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 
 #include <hardware/hardware.h>
 
diff --git a/mx6/hwcomposer/blit_ipu.h b/mx6/hwcomposer/blit_ipu.h
index 8f641a5..03ae5a8 100755
--- a/mx6/hwcomposer/blit_ipu.h
+++ b/mx6/hwcomposer/blit_ipu.h
@@ -1,3 +1,20 @@
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #ifndef _BLIT_IPU_H_
 #define _BLIT_IPU_H_
 
diff --git a/mx6/hwcomposer/hwc_common.cpp b/mx6/hwcomposer/hwc_common.cpp
index f0f147d..f5a4e2c 100755
--- a/mx6/hwcomposer/hwc_common.cpp
+++ b/mx6/hwcomposer/hwc_common.cpp
@@ -1,4 +1,20 @@
 
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 
 #include <hardware/hardware.h>
 
diff --git a/mx6/hwcomposer/hwc_common.h b/mx6/hwcomposer/hwc_common.h
index fd9311a..4b5ea9b 100755
--- a/mx6/hwcomposer/hwc_common.h
+++ b/mx6/hwcomposer/hwc_common.h
@@ -1,4 +1,20 @@
 
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #ifndef _HWC_FSL_H_
 #define _HWC_FSL_H_
 
diff --git a/mx6/hwcomposer/hwcomposer.cpp b/mx6/hwcomposer/hwcomposer.cpp
index 9615471..e2c049f 100755
--- a/mx6/hwcomposer/hwcomposer.cpp
+++ b/mx6/hwcomposer/hwcomposer.cpp
@@ -1,6 +1,4 @@
 /*
- * Copyright (C) 2010 The Android Open Source Project
- *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
@@ -14,6 +12,8 @@
  * limitations under the License.
  */
 
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #include <hardware/hardware.h>
 #include <hardware/overlay.h>
 
diff --git a/mx6/hwcomposer/output_device.cpp b/mx6/hwcomposer/output_device.cpp
index ceff1be..ffcaa02 100755
--- a/mx6/hwcomposer/output_device.cpp
+++ b/mx6/hwcomposer/output_device.cpp
@@ -1,4 +1,20 @@
 
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.*/
+
 #include <hardware/hardware.h>
 
 #include <fcntl.h>
-- 
1.8.0

